LCD.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * LCD.c
+ *
+ *  Created on: Feb 2, 2020
+ *      Author: helen
+ */
+
+#include "msp.h"
+#include "delay.h"
+#include "LCD.h"
+
+void init_LCD(void)                         //setup GPIO pins
+    {
+    P4-> SEL0 &= ~(DB4 | DB5 | DB6 | DB7);  //initialize P4 pins
+    P4-> SEL1 &= ~(DB4 | DB5 | DB6 | DB7);  //GPIO setup
+    P4-> DIR |=   (DB4 | DB5 | DB6 | DB7);  //set pins to output mode
+
+    P3-> SEL0 &= ~(RS | RW | EN);           //initialize P3 pins
+    P3-> SEL1 &= ~(RS | RW | EN);           //GPIO setup
+    P3-> DIR |= (RS | RW | EN);             //set pins to output mode
+
+    delay_ms(2,FREQ_48_MHZ);
+
+    P4-> OUT = 0x00;                        //clear outputs
+    P3-> OUT = 0x00;
+    delay_ms(100,FREQ_48_MHZ);              //power up delay
+    P4 -> OUT = 0x30;
+
+    delay_ms(30,FREQ_48_MHZ);               //power up delay
+    nybble();
+    delay_ms(10,FREQ_48_MHZ);
+    nybble();
+    delay_ms(10,FREQ_48_MHZ);
+    nybble();
+    delay_ms(10,FREQ_48_MHZ);
+
+    P4 -> OUT = 0x20;
+    nybble();
+
+    command_LCD(FUNC_SET);                  //4 bit, 2 line, 5x8 font
+    command_LCD(SET_CURSOR);                //set cursor
+    command_LCD(DISPLAY_ON);                //set display, cursor, blinking
+    command_LCD(CURSOR_RIGHT);              //set cursor direction
+
+    delay_ms(2,FREQ_48_MHZ);
+}
+
+void clear_LCD(void)                        //clears all characters from screen
+{
+    delay_ms(2, FREQ_48_MHZ);
+    command_LCD(CLR_DISP);
+    delay_ms(2, FREQ_48_MHZ);
+}
+
+void command_LCD(unsigned char command)     //reads instruction codes
+{
+    delay_ms(2,FREQ_48_MHZ);
+
+    P4-> OUT = command;                     //put instruction on OUT
+    P3-> OUT &= ~RS;                        //D/I = low --> Send Instruction
+    P3-> OUT &= ~RW;                        //R/W = low --> Write
+    nybble();
+
+    command = command << 4;                 //analyze next 4 bits
+    P4-> OUT = command;                     //shifted instruction code on OUT
+    nybble();
+
+    delay_ms(2,FREQ_48_MHZ);
+}
+
+void write_char_LCD(unsigned char letter)
+{
+    delay_ms(2,FREQ_48_MHZ);
+
+    P4 -> OUT = letter;                     //put letter data on OUT
+    P3-> OUT |= RS;                         //D/I = high --> Send Data
+    P3-> OUT &= ~RW;                        //R/W = low --> Write
+    nybble();
+
+    letter = letter << 4;                   //analyze next 4 bits
+    P4 -> OUT = letter;                     //shifted letter data on OUT
+    nybble();
+
+    delay_ms(2,FREQ_48_MHZ);
+}
+
+void write_string_LCD(char* string)
+{
+    while(*string)                          //loop until string completed
+    {
+        write_char_LCD(*string);            //write to LCD
+        string++;                           //count "up" to next char in string
+    }
+}
+
+void SetCursorLocation(char place)          //set cursor to DDRAM location as defined
```

Comment:
Change name to lcd_setCursorLocation. Make sure to change in the LCD.h and wherever its used